pandas/core/arrays/_mixins.py

```diff
@@ -194,3 +198,32 @@ def __getitem__(self, key):
 
     def _validate_getitem_key(self, key):
         return check_array_indexer(self, key)
+
+    @doc(ExtensionArray.fillna)
+    def fillna(self: _T, value=None, method=None, limit=None) -> _T:
+        value, method = validate_fillna_kwargs(value, method)
+
+        mask = self.isna()
+
+        # TODO: share this with EA base class implementation
+        if is_array_like(value):
+            if len(value) != len(self):
+                raise ValueError(
+                    f"Length of 'value' does not match. Got ({len(value)}) "
+                    f" expected {len(self)}"
+                )
+            value = value[mask]
+
+        if mask.any():
+            if method is not None:
+                func = missing.get_fill_func(method)
```

Comment:
`get_fill_func` returns the 1d versions pad and backfill, doesn't this mixin also need to also support 2d EAs?

Reply:
eventually yes.  but it wont break anything user-facing until we start backing DTBlock/TDBlock directly by DTA/TDA